go-controller/pkg/ovn/egressip.go

```diff
@@ -189,6 +189,18 @@ func (oc *DefaultNetworkController) reconcileEgressIP(old, new *egressipv1.Egres
 				return err
 			}
 			for _, namespace := range namespaces {
+				if util.IsNetworkSegmentationSupportEnabled() {
```

Comment:
you also check this in the `GetNetworkRoleForNamespace` function. Isn't it redundant then ?